tools/release_utils.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# Copyright 2023 The Vitess Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function checkGitState() {
+  git_status_output=$(git status --porcelain)
+  if [ "$git_status_output" == "" ]; then
+    	echo so much clean
+  else
+      echo "cannot do release with dirty git state"
+      exit 1
+  fi
+}
+
+function updateDockerReleaseScript () {
+  sed -i.bak -E "s/vt_base_version=.*/vt_base_version='$1'/g" $ROOT/docker/release.sh
+  rm -f $ROOT/docker/release.sh.bak
+}
+
+function checkoutNewBranch () {
+  branch_name=$1
+
+  current_branch=$BASE_BRANCH-$branch_name-1
+  git checkout -b $current_branch $BASE_REMOTE/$BASE_BRANCH
+  for (( i = 2; $? != 0; i++ )); do
+   current_branch=$BASE_BRANCH-$branch_name-$i
+   git checkout -b $current_branch $BASE_REMOTE/$BASE_BRANCH
+  done
+}
+
+function updateVersionGo () {
+
+  cat << EOF > ${ROOT}/go/vt/servenv/version.go
+/*
+Copyright 2022 The Vitess Authors.
```

Comment:
Today this would be 2023 🙂

We should instead make that dynamic:
```
YEAR=$(date +'%Y')
```

Reply:
Fixed in [6660312](https://github.com/vitessio/vitess/pull/12056/commits/6660312210281f65e7deffac7ac2169869949714)